src/test/java/com/atlassian/db/replica/api/AuroraClusterDiscoveryTest.java

```diff
@@ -66,6 +70,53 @@ void shouldDiscoverReplicaConnectionsWhenScaledUp() {
         assertThat(replicas).hasSize(2);
     }
 
+    @Test
+    void shouldUseClusterUri() {
+        auroraClusterMock
+            .scaleUp()
+            .scaleUp();
+
+        AuroraClusterDiscovery clusterDiscovery = AuroraClusterDiscovery.builder()
+            .replicaConnectionPerUrlProvider(replicaUrl -> () -> {
+                final Connection connection = mock(Connection.class);
+                final DatabaseMetaData databaseMetaData = mock(DatabaseMetaData.class);
+                when(databaseMetaData.getURL()).thenReturn(replicaUrl.toString());
+                when(connection.getMetaData()).thenReturn(databaseMetaData);
+                return connection;
+            })
+            .clusterUri(
+                "jdbc:postgresql://db-003.cluster-ro-xyz89.us-east-1.rds.amazonaws.com:5432/database")
```

Comment:
WHether this endpoint will be live forever? I wonder if external dependency might come with CI Flakyines?

Reply:
What external dependency?

Reply:
`db-003.cluster-ro-xyz89.us-east-1.rds.amazonaws.com:5432/database`

I haven't saw where you mocked mapping between
`db-003.cluster-ro-xyz89.us-east-1.rds.amazonaws.com:5432/database`
and
`apg-global-db-rpo-mammothrw-elephantro-n1.xyz89.us-east-1.rds.amazonaws.com:5432/database`